src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -29,45 +33,135 @@
 	public static void init() throws URISyntaxException, RelayException {
 		CONNECTION_URI = new URI(TestUtil.RELAY_NAMESPACE_URI + TestUtil.ENTITY_PATH);
 		tokenProvider = TokenProvider.createSharedAccessSignatureTokenProvider(TestUtil.KEY_NAME, TestUtil.KEY);
-		listener = new HybridConnectionListener(CONNECTION_URI, tokenProvider);
 		client = new HybridConnectionClient(CONNECTION_URI, tokenProvider);
-		listener.openAsync(Duration.ofSeconds(15)).join();
 	}
 	
-	@AfterClass
-	public static void cleanup() {
-		if (listener != null) {
-			listener.close();
-		}
+	@Test
+	public void openAndCloseTest() {
+		HybridConnectionListener listener = new HybridConnectionListener(CONNECTION_URI, tokenProvider);
+		AtomicBoolean onlineHandlerCalled = new AtomicBoolean(false);
```

Comment:
>AtomicBoolean [](start = 42, length = 13)

You could make these AtomicIntegers and then assert that they get called exactly one time (or however many it should be given the test).

Reply:
It only gets called once though, at least in this test.

---
In reply to: [261784456](https://github.com/Azure/azure-relay-java/pull/28#discussion_r261784456) [](ancestors = 261784456)

Reply:
How do you know it doesn't accidentally get called 2 or 3 times???

---
In reply to: [261791621](https://github.com/Azure/azure-relay-java/pull/28#discussion_r261791621) [](ancestors = 261791621,261784456)